linkerd/detect/src/lib.rs

```diff
@@ -142,3 +152,13 @@ where
         })
     }
 }
+
+// === impl DetectTimeout
+
+impl fmt::Display for DetectTimeout {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "Detection timed out after {:?}", self.0)
```

Comment:
very minor nit: in general, errors (including other errors added in this PR) tend to begin with lower case (so that you can say things like "foo error description: bar error description" without it looking weird):
```suggestion
        write!(f, "detection timed out after {:?}", self.0)
```